master/buildbot/process/build.py

```diff
@@ -327,9 +328,19 @@ def startBuild(self, build_status, workerforbuilder):
             yield self.buildPreparationFailure(ready_or_failure, "worker_prepare")
             if self.stopped:
                 self.buildFinished(["worker", "cancelled"], self.results)
+            elif isinstance(ready_or_failure, Failure) and ready_or_failure.check(interfaces.LatentWorkerFailedToSubstantiate):
```

Comment:
I think we need a CannotSubstanciate in order to distinguish transiant infra issues from issues that come from buildrequest